src/material/core/theming/_theming.scss

```
@@ -64,28 +64,110 @@
   @return rgba($color, if($opacity == null, opacity($color), $opacity));
 }
 
+// Validates the specified theme by ensuring that the optional color config defines
+// a primary, accent and warn palette. Returns the theme if no failures were found.
+@function _mat-validate-theme($theme) {
+  @if map_has_key($theme, color) {
+    $color: map_get($theme, color);
+    @if not map_get($color, primary) {
+      @error 'Theme does not define a valid "primary" palette.';
+    }
+    @else if not map_get($color, accent) {
+      @error 'Theme does not define a valid "accent" palette.';
+    }
+    @else if not map_get($color, warn) {
+      @error 'Theme does not define a valid "warn" palette.';
+    }
+  }
+  @return $theme;
+}
 
-// Creates a container object for a light theme to be given to individual component theme mixins.
-@function mat-light-theme($primary, $accent, $warn: mat-palette($mat-red)) {
+// Creates a light-themed color configuration from the specified
+// primary, accent and warn palettes.
+@function _mat-create-light-color-config($primary, $accent, $warn: null) {
   @return (
     primary: $primary,
     accent: $accent,
-    warn: $warn,
+    warn: if($warn != null, $warn, mat-palette($mat-red)),
     is-dark: false,
     foreground: $mat-light-theme-foreground,
     background: $mat-light-theme-background,
   );
 }
 
-
-// Creates a container object for a dark theme to be given to individual component theme mixins.
-@function mat-dark-theme($primary, $accent, $warn: mat-palette($mat-red)) {
+// Creates a dark-themed color configuration from the specified
+// primary, accent and warn palettes.
+@function _mat-create-dark-color-config($primary, $accent, $warn: null) {
   @return (
     primary: $primary,
     accent: $accent,
-    warn: $warn,
+    warn: if($warn != null, $warn, mat-palette($mat-red)),
     is-dark: true,
     foreground: $mat-dark-theme-foreground,
     background: $mat-dark-theme-background,
   );
 }
+
+// Creates a container object for a light theme to be given to individual component theme mixins.
+@function mat-light-theme($configOrPrimary, $accent: null, $warn: mat-palette($mat-red)) {
```

```suggestion
@function mat-light-theme($config-or-primary, $accent: null, $warn: mat-palette($mat-red)) {
```
(dash-case for Sass variables, here and elsewhere)

---

👍